frontend/src/queries/nodes/DataVisualization/DataVisualization.tsx

```diff
@@ -188,6 +188,7 @@ function InternalDataTableVisualization(props: DataTableVisualizationProps): JSX
                     <>
                         <HogQLQueryEditor
                             query={query.source}
+                            queryResponse={response ?? undefined}
```

Comment:
Not my finest hour, but I wasn't expecting the "saved query" functionality to be tied to the `HogQLQueryEditor` component. Not sure on a better way to do this without tearing apart this component and logic 🤷